src/box/lua/init.c

```diff
@@ -534,6 +535,8 @@ box_lua_init(struct lua_State *L)
 	lua_pop(L, 1);
 	luaopen_merger(L);
 	lua_pop(L, 1);
+	luaopen_key_estimator(L);
+	lua_pop(L, 1);
```

Comment:
Do you consider other possible names? It is multiset, right? Why not `key_multiset` or `keyring`? (But it is ofter used in the security context.) Or just `keyset` for simplicity. How about `multikey`? (Don't know, whether it has any relation to tarantool's multikey indexes, but if it does, it would be perfect.)

I would also engage @locker into the discussion. He ofter has good ideas regarding APIs.

Reply:
This is a HyperLogLog algorithm, so I think such names can be confusing.

Reply:
I misread the API: now I got that we add tuples into the multiset, not key_def objects. And it is not a container, it doesn't store the tuples.

Anyway, 'estimator' sounds too common here. What does it estimates? It is unclear from the name.

How about `key_def.hll` submodule?

Reply:
I don't want to call it `hll` because the Hyperloglog algorithm works with hashes, but the module works with tuples, so in my opinion the name should explain what it does in general, for example, *estimator*. The module estimates keys, so I guess it should be mentioned in its name too.

Could you explain why do want to make it a submosule of `key_def`?

Reply:
I don't like the names, either. TBH I don't understand why anyone would need this thing in Lua. (In C this might be helpful though - e.g. to estimate the number of unique elements in two merged Vinyl run files.) I think we should run this by the PM team.

Anyway, I would implement this module independently of box, tuples, or `key_def`. I think it should be named `hll` or `hyperloglog`, should be enabled with `require`, and should take hashes, like in the C API. To add tuples in HLL, we should add a function to the `key_def` module that takes a tuple and computes its hash. This way, the API would have clear names and would be more flexible, because we could add any hashable objects to it.

Reply:
I like your suggestions. Should I start implementing them or do I need to wait for a response from the PM team?

If the first, what do you think about the design of `key_def:hash()`? If we want to use this hash for the algorithm, we can't just use `tuple_hash()` because it returns 32-bit values. It also doesn't randomize the hash for integers, so the estimation for sets like {1, 2, 3} will be 1.
But at the same time I think that if `key_def:hash()` returns something other than the `tuple_hash()`, it might be confusing.

Reply:
PM @unera does not approve adding HyperLogLog to Tarantool Lua API.